akka-stream-tests/src/test/scala/akka/stream/scaladsl/SourceWithContextSpec.scala

```diff
@@ -112,8 +112,8 @@ class SourceWithContextSpec extends StreamSpec {
         .startContextPropagation(_.offset)
         .map(_.data)
         .statefulMapConcat(statefulFunction)
-        .endContextPropagation
-        .runWith(TestSink.probe[(String, Long)])
+        .toMat(TestSink.probe[(String, Long)])(Keep.right)
+        .run
```

Comment:
shouldn't `SourceWithContext` also have `runWith` for consistency with ordinary `Source`, making this typical case more convenient